src/main/java/io/temporal/internal/sync/POJOWorkflowImplMetadata.java

```diff
@@ -87,6 +88,11 @@ private POJOWorkflowImplMetadata(Class<?> implClass, boolean listener) {
     Class<?>[] interfaces = implClass.getInterfaces();
     for (int i = 0; i < interfaces.length; i++) {
       Class<?> anInterface = interfaces[i];
+
+      if (!Modifier.isPublic(anInterface.getModifiers())) {
+        continue;
+      }
```

Comment:
I know it would require a larger change, but ideally, we would throw an exception on nonpublic interface that is annotated with @ActivityInterface or @WorkflowInterface.

Reply:
done!